packages/core/src/__tests__/ActiveWorkflows.test.ts

```diff
@@ -0,0 +1,376 @@
+					getPollFunctions,
+				);
+
+				// Get the executeTrigger function that was registered
+				const registerCronCall = scheduledTaskManager.registerCron.mock.calls[0];
+				const executeTrigger = registerCronCall[2] as () => Promise<void>;
+
+				// Execute the trigger function to simulate a regular poll
+				await executeTrigger();
+
+				expect(triggersAndPollers.runPoll).toHaveBeenCalledTimes(2);
+				expect(pollFunctions.__emit).not.toHaveBeenCalled();
+				expect(pollFunctions.__emitError).toHaveBeenCalledWith(error);
+			});
+		});
+	});
+
+	describe('remove()', () => {
+		beforeEach(() => {
+			workflow.getTriggerNodes.mockReturnValue([triggerNode]);
+			workflow.getPollNodes.mockReturnValue([]);
+		});
+
+		it('should remove an active workflow', async () => {
+			await activeWorkflows.add(
+				workflowId,
+				workflow,
+				additionalData,
+				mode,
+				activation,
+				getTriggerFunctions,
+				getPollFunctions,
+			);
+
+			const result = await activeWorkflows.remove(workflowId);
+
+			expect(result).toBe(true);
+			expect(activeWorkflows.isActive(workflowId)).toBe(false);
+			expect(scheduledTaskManager.deregisterCrons).toHaveBeenCalledWith(workflowId);
+			expect(triggerResponse.closeFunction).toHaveBeenCalled();
+		});
+
+		it('should return false when removing non-existent workflow', async () => {
+			const result = await activeWorkflows.remove('non-existent');
+
+			expect(result).toBe(false);
+			expect(scheduledTaskManager.deregisterCrons).not.toHaveBeenCalled();
+		});
+
+		it('should handle TriggerCloseError when closing trigger', async () => {
+			const triggerCloseError = new TriggerCloseError(triggerNode, { level: 'warning' });
+			(triggerResponse.closeFunction as jest.Mock).mockRejectedValueOnce(triggerCloseError);
+
+			await activeWorkflows.add(
+				workflowId,
+				workflow,
+				additionalData,
+				mode,
+				activation,
+				getTriggerFunctions,
+				getPollFunctions,
+			);
+
+			const result = await activeWorkflows.remove(workflowId);
+
+			expect(result).toBe(true);
+			expect(activeWorkflows.isActive(workflowId)).toBe(false);
+			expect(triggerResponse.closeFunction).toHaveBeenCalled();
+			expect(errorReporter.error).toHaveBeenCalledWith(triggerCloseError, {
+				extra: { workflowId },
+			});
+		});
+
+		it('should throw WorkflowDeactivationError when closeFunction throws regular error', async () => {
+			const error = new Error('Close function failed');
+			(triggerResponse.closeFunction as jest.Mock).mockRejectedValueOnce(error);
+
+			await activeWorkflows.add(
+				workflowId,
+				workflow,
+				additionalData,
+				mode,
+				activation,
+				getTriggerFunctions,
+				getPollFunctions,
+			);
+
+			await expect(activeWorkflows.remove(workflowId)).rejects.toThrow(
+				`Failed to deactivate trigger of workflow ID "${workflowId}": "Close function failed"`,
+			);
+
+			expect(triggerResponse.closeFunction).toHaveBeenCalled();
+			expect(errorReporter.error).not.toHaveBeenCalled();
```

Comment:
Should `isActive` be true or false now?

Reply:
it's actually unclear. I've posted [here](https://n8nio.slack.com/archives/C069HS026UF/p1734530918502139) for clarification.  
Once we have consensus, we can create a new ticket and new PR to address this.